### PR TITLE
[FIX] website_sale : make the cart icon visible

### PR DESCRIPTION
**Pull request by anso-odoo**
Cart icon in the header was invisible in some themes

task-3044936

### CODE REVIEW

**Comment by stefanorigano** [comment]
Hello @odoo/rd-website , before proceeding further with this... 

@anso-odoo solution solves the problem and show the cart icon for themes that mistakenly "hide" it.
The puzzling part is that there is nothing in the default theme that set the icon to "visible"... but still, it is mysteriously shown correctly.

to replicate:
default theme (contact menu) -> Icon is shown
expert theme (contact menu) -> Icon is hidden

There are maybe discrepancies in the way how we call these templates **when using themes**?
Or maybe it's a `t-cache` issue?

.. or maybe I'm missing something 😅 ?



Thanks

File: addons/website_sale/views/templates.xml [hunk 1]
```diff
@@ -37,6 +37,7 @@
     <template id="template_header_hamburger" inherit_id="website.template_header_hamburger">
         <xpath expr="//t[@t-foreach='website.menu_id.child_id']" position="after">
             <t t-call="website_sale.header_cart_link">
+                <t t-set="_icon" t-value="True"/>
```
Comment (rdeodoo) [inline addons/website_sale/views/templates.xml:40]: The code is clearly showing that the templates are designed to either show the icon (fa-xx) or a text instead ('My Cart').
If you pass the value to `True`, the icon is indeed visible, but that's breaking the layout that was trying to be achieved: Some templates are showing the text, other are showing the icon.

**Review by rdeodoo — CHANGES_REQUESTED** [review]
That's indeed a `t-cache` issue.
I did not yet finished my commit message nor the analysis but here is the correct fix: https://github.com/odoo/odoo/pull/104232
I'll check the theme behavior you are speaking about and keep updated about it all by tomorrow

**Comment by rdeodoo** [comment]
I couldn't reproduce the mismatch between community and themes regarding the use of the icon/text.
Themes should not (and do not from what I was able to test) impact the cart icon layout, this is defined at the template header level which is not impacted by the themes (they indeed change which template is used but not the template itself).

If you have a step to reproduce I can investigate deeper but for now, I make the fix at https://github.com/odoo/odoo/pull/104232 which is ready if you want to confirm it solves the issue.

Side note but not related to this directly (I found it while investigating your report): I made this PR in themes https://github.com/odoo/design-themes/pull/610
Basically, the default themes website created on runbot through the `test_themes` app were missing the call to `_post_copy()`, meaning that the header and footer template were not activated as it should have been (on top of other stuff like ripple etc).

**Comment by stefanorigano** [comment]
Thanks @rdeodoo , it may be one either the `t-cache` or `_post_copy()` indeed, I can't see any other reasons.
Btw, to replicate just switch to "expert" theme on runbot

@anso-odoo ,  meanwhile we wait for https://github.com/odoo/design-themes/pull/610 to be merged, could you test locally https://github.com/odoo/odoo/pull/104232 and see if it solves the issue?
Also, it would be good to know of the issue affects runbot only or local instances too.

**Comment by rdeodoo** [comment]
Oh, yeah, that's because of the bug that I am fixing with https://github.com/odoo/design-themes/pull/610
When you switch to the expert theme, it actually is not really in "Contact header" mode, you can easily see it if you go to the Debug > Views list: there is not a single header template enabled for any theme websites, because those websites were created through `test_themes` which is not calling the `post_copy()`.

So if you then, on Expert theme, change the template header to something else and then reselect "Contact" header template, the mismatch/weird behavior will be gone, it will correctly use the template and behaves as the default website / a website without theme would do.

**Comment by stefanorigano** [comment]
Great, thanks @rdeodoo 👌
I'm closing this task.

cc. @anso-odoo
